### PR TITLE
chore: update to Nuxt 3.10.2

### PR DESCRIPTION
**Pull request by userquin**
This PR requires a review and includes:
- update to nuxt 3.10.2
- update to i18n 8.1.1 (some types moved to the module: `LocaleObject` and `Directions` from `@nuxtjs/i18n`): update all required types and moved the configuration to the new i18n options
- update a lot of pages that now requires `isHydrated` to avoid hydration missmatch (also some setting pages updated to remove `isHydrated`): talking with @patak-dev we need to review the `isHydrated`, but I guess we cannot remove it since we've `app.keepAlive`
- pwa not yet tested: updated pwa plugin to latest `0.19.0`, I will do a full review tmr
- ~~module resolution set to `node`: we need a fix for `@vueuse/gesture`, cannot be used with `Bundler`: @danielroe sent a PR to the repo https://github.com/vueuse/gesture/pull/25 (@antfu maybe you can do something on the repo: I have sent also a PR to fix types in `@vueuse/motion`, we don't use the nuxt module but it can be a problem if we enable it here https://github.com/vueuse/motion/pull/175)~~
- vitest version fixed, the same using `@nuxt/test-utils`  (`1.2.2`)
- changed all `process.client`, `process.server` and `process.dev` with `import.meta.client`, `import.meta.server` and `import.meta.dev` respectivelly, we also need to review `process.test` (it is configured in `vite.define`) 
- rn `typescheck` script should be fine, on my local works

The problem with the browser was the `AccountHoverWrapper.vue`, we need to defer the account fetch until visible via interceptor observer when providing only the account handle.

Also fixed the paginator module, changing the state to shallowRef to avoid multiples calls from the watch effect.

We also need to review page lifecycle again, switching between tabs reloading the page when back to elk tab.

supersedes #2543

### CODE REVIEW

File: components/account/AccountHoverWrapper.vue [hunk 1]
```diff
@@ -11,16 +16,60 @@ const props = defineProps<{
...
+useIntersectionObserver(
+  hoverCard,
+  ([{ intersectionRatio }]) => {
+    targetIsVisible.value = intersectionRatio <= 0.75
+  },
+)
+
+watch(
+  () => [props.account, props.handle, targetIsVisible.value] satisfies PropsType,
+  async ([newAccount, newHandle, newVisible], oldProps) => {
```
Comment (userquin) [inline components/account/AccountHoverWrapper.vue:32]: remove `async`

File: .vscode/settings.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
     "unstorage"
   ],
   "editor.codeActionsOnSave": {
-    "source.fixAll.eslint": true
+    "source.fixAll.eslint": "explicit"
```
Comment (userquin) [inline .vscode/settings.json:9]: revert this change

File: components/account/AccountHoverWrapper.vue [hunk 3]
```diff
@@ -1,5 +1,10 @@
 <script setup lang="ts">
 import type { mastodon } from 'masto'
+import type { Ref } from 'vue'
+import { useAccountByHandle } from '~/composables/cache'
```
Comment (userquin) [inline components/account/AccountHoverWrapper.vue:4]: remove this import and review types

File: components/account/AccountHoverWrapper.vue [hunk 4]
```diff
@@ -11,16 +16,60 @@ const props = defineProps<{
   disabled?: boolean
 }>()
 
-const account = computed(() => props.account || (props.handle ? useAccountByHandle(props.handle!) : undefined))
+const hoverCard = ref()
+const targetIsVisible = ref(false)
+const account = ref<AccountType>(props.account)
```
Comment (userquin) [inline components/account/AccountHoverWrapper.vue:21]: create a computed with `toValue`?

File: components/account/AccountHoverWrapper.vue [hunk 5]
```diff
@@ -11,16 +16,60 @@ const props = defineProps<{
...
+      account.value = newAccount
+      return
+    }
+
+    if (!newVisible)
+      return
+
+    if (newHandle) {
+      const [_oldAccount, oldHandle, _oldVisible] = oldProps ?? [undefined, undefined, false]
+      if (!oldHandle || newHandle !== oldHandle || !account.value) {
```
Comment (userquin) [inline components/account/AccountHoverWrapper.vue:43]: review this with the new computed

File: components/status/StatusActionsMore.vue [hunk 6]
```diff
@@ -88,7 +88,7 @@ async function deleteAndRedraft() {
   }) !== 'confirm')
     return
 
-  if (process.dev) {
+  if (import.meta.env.DEV) {
```
Comment (danielroe) [inline components/status/StatusActionsMore.vue:91]:
```suggestion
  if (import.meta.dev) {
```

File: composables/cache.ts [hunk 7]
```diff
@@ -5,7 +5,7 @@ const cache = new LRUCache<string, any>({
   max: 1000,
 })
 
-if (process.dev && process.client)
+if (import.meta.env.DEV && import.meta.client)
```
Comment (danielroe) [inline composables/cache.ts:8]:
```suggestion
if (import.meta.dev && import.meta.client)
```

File: composables/dialog.ts [hunk 8]
```diff
@@ -56,7 +56,7 @@ export async function openPublishDialog(draftKey = 'dialog', draft?: Draft, over
     if (overwrite && !isEmptyDraft(currentUserDrafts.value[draftKey])) {
       // TODO overwrite warning
       // TODO don't overwrite, have a draft list
-      if (process.dev) {
+      if (import.meta.env.DEV) {
```
Comment (danielroe) [inline composables/dialog.ts:59]:
```suggestion
      if (import.meta.dev) {
```

File: composables/masto/publish.ts [hunk 9]
```diff
@@ -94,7 +94,7 @@ export function usePublish(options: {
       ...(isGlitchEdition.value ? { 'content-type': 'text/markdown' } : {}),
     } as mastodon.rest.v1.CreateStatusParams
 
-    if (process.dev) {
+    if (import.meta.env.DEV) {
```
Comment (danielroe) [inline composables/masto/publish.ts:97]:
```suggestion
    if (import.meta.dev) {
```

File: plugins/setup-head-script.server.ts [hunk 10]
```diff
@@ -16,7 +16,7 @@ export default defineNuxtPlugin(() => {
   if (!settings) { return }
 
   const html = document.documentElement
-  ${process.dev ? 'console.log({ settings })' : ''}
+  ${import.meta.env.DEV ? 'console.log({ settings })' : ''}
```
Comment (danielroe) [inline plugins/setup-head-script.server.ts:19]:
```suggestion
  ${import.meta.dev ? 'console.log({ settings })' : ''}
```

File: modules/tauri/runtime/nitro.client.ts [hunk 11]
```diff
@@ -37,7 +37,7 @@ export default defineNuxtPlugin(async () => {
   const config = useRuntimeConfig()
 
   const h3App = createApp({
-    debug: process.dev,
+    debug: import.meta.env.DEV,
```
Comment (danielroe) [inline modules/tauri/runtime/nitro.client.ts:40]:
```suggestion
    debug: import.meta.dev,
```

**Comment by Tahul** [comment]
Hey @danielroe ;

Just noticed your PR on @vueuse/gesture; I just released `2.0.0`.

Could you let me know it that adresses your current issues with the package?

**Comment by userquin** [comment]
> Could you let me know if that adresses your current issues with the package?

Yes, the new version fixes the problem with Bundler 

thx ❤️

**Comment by userquin** [comment]
I'm going to split this PR in 5:
- `@vueuse/gesture 2.0.0`
- `vue v3.4`
- `nuxt v3.10.3`
- `i18n v8.1.1`
- bump dependencies (tiptap and floating-vue)

**Comment by patak-dev** [comment]
Closing, as https://github.com/elk-zone/elk/pull/2610 has now been merged.
Awesome work with these updates @userquin!
